modules/apps/login/login-web/src/main/java/com/liferay/login/web/internal/portlet/action/SetPasswordMVCActionCommand.java

```diff
@@ -0,0 +1,326 @@
+	protected void doProcessAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
+		throws Exception {
+
+		Ticket ticket = getTicket(actionRequest);
+
+		actionRequest.setAttribute(WebKeys.TICKET, ticket);
+
+		String cmd = ParamUtil.getString(actionRequest, Constants.CMD);
+
+		if (Validator.isNull(cmd)) {
+			if (ticket != null) {
+				User user = _userLocalService.getUser(ticket.getClassPK());
+
+				_userLocalService.updatePasswordReset(user.getUserId(), true);
+			}
+
+			User user = _portal.getUser(actionRequest);
+
+			if ((user != null) && _isUserDefaultAdmin(user)) {
+				String reminderQueryAnswer = user.getReminderQueryAnswer();
+
+				if (Validator.isNotNull(reminderQueryAnswer) &&
+					reminderQueryAnswer.equals(
+						WorkflowConstants.LABEL_PENDING)) {
+
+					actionRequest.setAttribute(
+						WebKeys.TITLE_SET_PASSWORD, "set-password");
+				}
+			}
+		}
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		try {
+			updatePassword(actionRequest, actionResponse, themeDisplay, ticket);
+
+			String redirect = ParamUtil.getString(
+				actionRequest, WebKeys.REFERER);
+
+			if (Validator.isNotNull(redirect)) {
+				redirect = _portal.escapeRedirect(redirect);
+			}
+
+			if (Validator.isNull(redirect)) {
+				redirect = themeDisplay.getPathMain();
+			}
+
+			actionResponse.sendRedirect(redirect);
+		}
+		catch (Exception exception) {
+			if (exception instanceof UserPasswordException) {
+				SessionErrors.add(
+					actionRequest, exception.getClass(), exception);
+			}
+			else if (exception instanceof NoSuchUserException ||
+					 exception instanceof PrincipalException) {
+
+				SessionErrors.add(actionRequest, exception.getClass());
+			}
+
+			_portal.sendError(exception, actionRequest, actionResponse);
+		}
+	}
+
+	protected Ticket getTicket(ActionRequest actionRequest)
+		throws PortalException {
+
+		String ticketId = ParamUtil.getString(actionRequest, "ticketId");
+
+		String ticketKey = ParamUtil.getString(actionRequest, "ticketKey");
+
+		if (Validator.isNull(ticketId) || Validator.isNull(ticketKey)) {
+			return null;
+		}
+
+		try {
+			Ticket ticket = _ticketLocalService.fetchTicket(
+				GetterUtil.getLong(ticketId));
+
+			if ((ticket == null) ||
+				(ticket.getType() != TicketConstants.TYPE_PASSWORD)) {
+
+				return null;
+			}
+
+			String encryptedTicketKey = PasswordEncryptorUtil.encrypt(
+				ticketKey, ticket.getKey());
+
+			if (!ticket.isExpired() &&
+				encryptedTicketKey.equals(ticket.getKey())) {
+
+				return ticket;
+			}
+
+			_ticketLocalService.deleteTicket(ticket);
+		}
+		catch (Exception exception) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(exception);
+			}
+		}
+
+		return null;
+	}
+
+	protected boolean isValidatePassword(
+		HttpServletRequest httpServletRequest) {
+
+		HttpSession httpSession = httpServletRequest.getSession();
+
+		Boolean setupWizardPasswordUpdated = (Boolean)httpSession.getAttribute(
```

Comment:
Use GetterUtil.getBoolean instead: 
```
Boolean setupWizardPasswordUpdated = GetterUtil.getBoolean(httpSession.getAttribute(
  WebKeys.SETUP_WIZARD_PASSWORD_UPDATED));
```

It will simplify your if logic to just:

```
if (setupWizardPasswordUpdated) {
     ...
}
```